From 4342efdd1e3b2fe262bae7ced55abeeb58be7620 Mon Sep 17 00:00:00 2001
From: "Wesley W. Terpstra" <wesley@sifive.com>
Date: Mon, 20 Jun 2016 18:16:32 -0700
Subject: [PATCH 02/10] riscv: linker script bug when adding more options

  LD      init/built-in.o
`.L11' referenced in section `__bug_table' of crypto/built-in.o: defined in discarded section `.exit.text' of crypto/built-in.o
/home/terpstra/linux-flow/linux-4.6.2/Makefile:949: recipe for target 'vmlinux' failed

... all the other ports have this magic incantation since 2012.
---
 arch/riscv/kernel/vmlinux.lds.S | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/riscv/kernel/vmlinux.lds.S b/arch/riscv/kernel/vmlinux.lds.S
index e60e39a..97e2da7 100644
--- a/arch/riscv/kernel/vmlinux.lds.S
+++ b/arch/riscv/kernel/vmlinux.lds.S
@@ -18,6 +18,15 @@ SECTIONS
 	HEAD_TEXT_SECTION
 	INIT_TEXT_SECTION(PAGE_SIZE)
 	INIT_DATA_SECTION(16)
+	/* we have to discard exit text and such at runtime, not link time */
+	.exit.text :
+	{
+		EXIT_TEXT
+	}
+	.exit.data :
+	{
+		EXIT_DATA
+	}
 	PERCPU_SECTION(L1_CACHE_BYTES)
 	__init_end = .;
 
-- 
2.10.2

